staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -711,13 +711,23 @@ func (w *watchCache) getAllEventsSinceLocked(resourceVersion uint64) (*watchCach
 	}
 
 	if resourceVersion == 0 {
-		// resourceVersion = 0 means that we don't require any specific starting point
-		// and we would like to start watching from ~now.
-		// However, to keep backward compatibility, we additionally need to return the
-		// current state and only then start watching from that point.
-		//
-		// TODO: In v2 api, we should stop returning the current state - #13969.
-		return w.getIntervalFromStoreLocked()
+		if opts.SendInitialEvents == nil {
+			// resourceVersion = 0 means that we don't require any specific starting point
+			// and we would like to start watching from ~now.
+			// However, to keep backward compatibility, we additionally need to return the
+			// current state and only then start watching from that point.
+			//
+			// TODO: In v2 api, we should stop returning the current state - #13969.
+			return w.getIntervalFromStoreLocked()
+		}
+		if *opts.SendInitialEvents {
+			// this is a programmer error
+			return nil, errors.NewInternalError(fmt.Errorf("opts.SendInitialEvents cannot be set to true"))
```

Comment:
Because it should be handled by `cacher.go:614: opts.SendInitialEvents != nil && *opts.SendInitialEvents`. We could get rid of `614` line and move the logic to `getAllEventsSinceLocked`, wdyt?

Reply:
Yeah - let's move this logic here...